### PR TITLE
Bass.ChannelGetLevelLeft/Right returns negative values if paused

### PR DESCRIPTION
**Pull request by swoolcock**
Added a check in TrackBass to prevent amplitudes being updated to crazy values if the track is paused.
Related to: https://github.com/ppy/osu/pull/1243
Possibly also a fix for: https://github.com/ppy/osu/issues/1284

### CODE REVIEW

**Comment by gabixdev** [comment]
Confirmed, fixes #1284.

File: osu.Framework/Audio/Track/TrackBass.cs [hunk 1]
```diff
@@ -101,15 +101,18 @@ public override void Update()
...
+            var rightChannel = Bass.ChannelGetLevelRight(activeStream) / 32768f;
+
+            if (leftChannel >= 0 && rightChannel >= 0)
+            {
+                currentAmplitudes.LeftChannel = leftChannel;
+                currentAmplitudes.RightChannel = rightChannel;
+
+                float[] tempFrequencyData = new float[256];
+                Bass.ChannelGetData(activeStream, tempFrequencyData, (int)DataFlags.FFT512);
+                currentAmplitudes.FrequencyAmplitudes = tempFrequencyData;
```
Comment (peppy) [inline osu.Framework/Audio/Track/TrackBass.cs]: we probably want these zero'd out when in this state

File: osu.Framework/Audio/Track/TrackBass.cs [hunk 2]
```diff
@@ -101,15 +101,18 @@ public override void Update()
...
-            tempLevel = Bass.ChannelGetLevelRight(activeStream) / 32768f;
-            currentAmplitudes.RightChannel = tempLevel == -1 ? 1 : tempLevel;
-
-            float[] tempFrequencyData = new float[256];
-            Bass.ChannelGetData(activeStream, tempFrequencyData, (int)DataFlags.FFT512);
-            currentAmplitudes.FrequencyAmplitudes = tempFrequencyData;
+            var leftChannel = Bass.ChannelGetLevelLeft(activeStream) / 32768f;
+            var rightChannel = Bass.ChannelGetLevelRight(activeStream) / 32768f;
+
+            if (leftChannel >= 0 && rightChannel >= 0)
```
Comment (peppy) [inline osu.Framework/Audio/Track/TrackBass.cs]: tried using IsPlaying instead, and only doing the `GetLevel`s if so? Would avoid two (potentially expensive) bass calls when we aren't running, and likely be more correct behaviour.

Comment (swoolcock) [reply osu.Framework/Audio/Track/TrackBass.cs]: Can I use `isPlayed`?  Based on the docs, I assume `true` means paused or stopped.

Comment (swoolcock) [reply osu.Framework/Audio/Track/TrackBass.cs]: Other way around, rather.

**Pull request merged**